src/items.rs

```diff
@@ -506,27 +508,62 @@ impl<'a> FmtVisitor<'a> {
         // without spans for the comment or parens, there is no chance of
         // getting it right. You also don't get to put a comment on self, unless
         // it is explicit.
-        if args.len() >= min_args {
+        if args.len() >= min_args || variadic {
             let comment_span_start = if min_args == 2 {
                 span_after(span, ",", self.codemap)
             } else {
                 span.lo
             };
 
+            enum ArgumentKind<'a> {
+                Regular(&'a ast::Arg),
+                Variadic,
+            }
+
+            let variadic_arg = if variadic {
+                Some(ArgumentKind::Variadic)
+            } else {
+                None
+            };
+            let variadic_start = if variadic {
+                let span = codemap::mk_sp(args.last().unwrap().ty.span.hi, span.hi);
+                span_after(span, "...", self.codemap) - BytePos(1)
+            } else {
+                BytePos(0)
+            };
```

Comment:
Could you add the start byte pos to the Variadic enum variant? Then you only need one if/else here and you don't need the magic 0 for variadic_start